pkg/controllers/subnet/subnet_poll.go

```diff
@@ -0,0 +1,229 @@
+package subnet
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/vmware/vsphere-automation-sdk-go/services/nsxt/model"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/vmware-tanzu/nsx-operator/pkg/apis/vpc/v1alpha1"
+	"github.com/vmware-tanzu/nsx-operator/pkg/nsx/services/common"
+)
+
+var ticker = time.NewTicker(10 * time.Minute)
+
+// pollSharedSubnets periodically polls NSX for shared subnet status updates
+// It runs in a separate goroutine and polls every 10 minutes.
+// The polling can be stopped by sending a value to the stopCh channel.
+func (r *SubnetReconciler) pollSharedSubnets(stopCh chan bool) {
+	defer ticker.Stop()
+	for {
+		select {
+		case <-ticker.C:
+			r.pollAllSharedSubnets()
+		case <-stopCh:
+			log.Info("Stopping shared Subnet polling")
+			return
+		}
+	}
+}
+
+// pollAllSharedSubnets polls NSX for all shared subnets.
+// It uses the field indexer to find all Subnet resources with the associated_resource annotation.
+// It groups subnets by associatedResource to avoid redundant NSX API calls.
+// For each unique associatedResource, it gets the NSX subnet and status only once,
+// then updates all related Subnet CRs with the same information.
+// It also removes subnets from the queue if they no longer exist or are being deleted.
+func (r *SubnetReconciler) pollAllSharedSubnets() {
+	// Group namespacedNames by associatedResource
+	resourceMap := make(map[string][]types.NamespacedName)
+	ctx := context.Background()
+
+	// Use the field indexer to find all Subnet resources with the associated_resource annotation
+	subnetList := &v1alpha1.SubnetList{}
+	err := r.Client.List(ctx, subnetList, client.MatchingFields{
+		"metadata.annotations." + common.AnnotationAssociatedResource: "true",
+	})
+	if err != nil {
+		log.Error(err, "Failed to list Subnets with associated_resource annotation")
+		return
+	}
+
+	// Group subnets by associatedResource
+	for _, subnet := range subnetList.Items {
+		associatedResource := subnet.Annotations[common.AnnotationAssociatedResource]
+		namespacedName := types.NamespacedName{
+			Namespace: subnet.Namespace,
+			Name:      subnet.Name,
+		}
+		resourceMap[associatedResource] = append(resourceMap[associatedResource], namespacedName)
+	}
+
+	// Process each unique associatedResource
+	for associatedResource, namespacedNames := range resourceMap {
+		ctx := context.Background()
```

Comment:
This can be removed as we already have ctx defined at line 45?

Reply:
fixed